zone/lua_client.cpp

```diff
@@ -1761,9 +1766,27 @@ void Lua_Client::QuestReward(Lua_Mob target, luabind::adl::object reward) {
 
 	cur = reward["faction"];
 	if (luabind::type(cur) != LUA_TNIL) {
-		try {
-			faction = luabind::object_cast<bool>(cur);
-		} catch (luabind::cast_failed &) {
+		// if it's a table it will be {faction, faction_mod}
+		if (luabind::type(cur) == LUA_TTABLE) {
+			auto item = cur[1];
+			if (luabind::type(item) != LUA_TNIL) {
+				try {
```

Comment:
Should be able to just check if type is `LUA_TNUMBER` here and do the cast without the try/catch. The parser will catch any exceptions as quest errors anyway for bad inputs which might be more desirable than silently failing

Reply:
I was copying the pattern we were already doing.

Reply:
I think it might be scope creep to rewrite this, the function does this with other stuff and rewriting the whole function is probably a good idea, but that's not implementing this feature.